data_broker.js

```diff
@@ -151,7 +155,12 @@ class DataBroker extends EventTarget {
       return;
     }
 
-    if (!this._header) {
+    const alignmentUrlChanged = this.alignmentUrl !== this._lastAlignmentUrl;
+
+    if (alignmentUrlChanged) {
+
```

Comment:
emit `alignment-data-request` here

Reply:
Good point! That is related to the one above (why I emit the event in the alignmentUrl setter). So I will explain here together. 

The line of code `this._alignmentUrl !== _` looks the same as `if (alignmentUrlChanged) `. The reason I emitted the event earlier is that I tried to emit the event `alignment-data-request` in the ` if (alignmentUrlChanged) `, but the old chart flashes before the loading indicator shows. Emitting the event earlier solved this problem.

Reply:
Why does it flash in one case and not the other?

Reply:
I think `_doUpdate` is an async function. This might cause the timing issue. 

What's your thought here?

Reply:
That's probably the essence of the problem. I'm concerned with the current solution though. It's not at all obvious why that code needs to be there. And it would be very easy for a future programmer to move it and not realize it creates a flashing problem. The data broker shouldn't be concerned with controlling when to show and hide UI components.

If you need tighter control over the UI, I think you should control that from index.html or the component itself. For example, you could hook in to the file selection event, or maybe create a new event. Think about it and go with what you think is cleanest then I'll take a look.